src/operator/quantization/quantize_graph_pass.cc

```diff
@@ -102,28 +102,57 @@ std::vector<NodeEntry> OfflineParams(std::vector<NodeEntry>&& outputs,
+  const auto& op = node->op();
+  Context ctx = Context::Create(static_cast<Context::DeviceType>(dev_type), 0);
+  FCompute fcompute = common::GetFCompute<FCompute>(op, "FCompute", ctx);
+  FComputeEx fcomp_ex = common::GetFCompute<FComputeEx>(op, "FComputeEx", ctx);
+  FStatefulCompute fcomputestateful =
+      common::GetFCompute<FStatefulCompute>(op, "FStatefulCompute", ctx);
+  FStatefulComputeEx fcomputestateful_ex =
+      common::GetFCompute<FStatefulComputeEx>(op, "FStatefulComputeEx", ctx);
+  return (fcompute != nullptr || fcomp_ex != nullptr ||
+          fcomputestateful != nullptr || fcomputestateful_ex != nullptr);
+}
+
+inline NodePtr NeedQuantize(
+    NodePtr node, const std::unordered_set<std::string>& excluded_nodes,
+    const std::unordered_set<std::string>& excluded_ops,
+    const int& dev_type) {
   std::unordered_map<NodePtr, NodePtr> quantized_node;
   static auto& quantized_op_map = Op::GetAttr<mxnet::FQuantizedOp>("FQuantizedOp");
   static auto& fexec_type = nnvm::Op::GetAttr<FExecType>("FExecType");
   const auto& op = node->op();
 
   if (op && quantized_op_map.count(op)) {
     bool need = true;
-    if (excluded_nodes.count(node->attrs.name)) {
+    // If the quantized node is not registered with a computation function, the node
+    // will be excluded automatically.
+    auto q_ptr = quantized_op_map[node->op()];
+    auto qnode = q_ptr(node->attrs);
+    if (!isRegistered(qnode, dev_type)) {
+      LOG(INFO) << "Neither FCompute nor FComputeEx registered, " << node->op()->name
+                << " excluded automatically.";
       need = false;
-    } else if (!node->attrs.subgraphs.empty()) {
-      ExecType exec_type = fexec_type.count(op) ? fexec_type[op](node->attrs) : ExecType::kSync;
-      if (exec_type != ExecType::kSubgraphExec) {
-        // This is a fused subgraph node, try to match inner node.
-        CHECK_EQ(node->attrs.subgraphs.size(), 1);
-        auto subgraph_sym = node->attrs.subgraphs[0];
-        DFSVisit(subgraph_sym->outputs, [&](const nnvm::NodePtr& n) {
-          if (n->is_variable()) return;
-          if (excluded_nodes.count(n->attrs.name)) {
-            need = false;
-          }
-        });
+    } else {
+      if (excluded_nodes.count(node->attrs.name) ||
+          excluded_ops.count(node->op()->name)) {
+        need = false;
+      } else if (!node->attrs.subgraphs.empty()) {
+        ExecType exec_type = fexec_type.count(op) ? fexec_type[op](node->attrs) : ExecType::kSync;
+        if (exec_type != ExecType::kSubgraphExec) {
+          // This is a fused subgraph node, try to match inner node.
+          CHECK_EQ(node->attrs.subgraphs.size(), 1);
+          auto subgraph_sym = node->attrs.subgraphs[0];
+          DFSVisit(subgraph_sym->outputs, [&](const nnvm::NodePtr& n) {
+            if (n->is_variable()) return;
+            if (excluded_nodes.count(n->attrs.name) ||
+                excluded_ops.count(node->op()->name)) {
```

Comment:
Fused op is a new op, so we shouldn't check its inner node.

Reply:
I found we cannot exclude fused conv layers when setting`excluded_op_names=['Convolution']`. Is it necessary to check the inner node here?